packages/next-app/src/app/internalmanager/(components)/TokenTable.tsx

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-console */
+import { useAccount, useNetwork } from "wagmi";
+
+import { Button } from "#/components";
+import Table from "#/components/Table";
+import { impersonateWhetherDAO, pools } from "#/lib/gql";
+import { tokenDictionary } from "#/utils/getTokenInfo";
+import { writeWithdrawInternalBalance } from "#/wagmi/withdrawInternalBalance";
+
+export enum UserBalanceOpKind {
+  DEPOSIT_INTERNAL,
+  WITHDRAW_INTERNAL,
+  TRANSFER_INTERNAL,
+  TRANSFER_EXTERNAL,
+}
+
+export function TokenTable() {
+  const { chain } = useNetwork();
+
+  let { address } = useAccount();
+  address = impersonateWhetherDAO(chain?.id.toString() || "1", address);
+
+  const { data } = pools.gql(chain!.id.toString()).useInternalBalance({
+    userAddress: address!.toLowerCase(),
+  });
```

Comment:
Very error prone 😬 would be nice to add some error handling when these are empty or create a task to work on it.

Reply:
Done. [Issue here](https://linear.app/bleu-llc/issue/BAL-266/error-handling-for-empty-address)